### PR TITLE
feat: upgrade to actix-web 4

### LINKED ISSUE #386: Upgrade to actix-web 4
┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/CONSVC-1783)

### PR DESCRIPTION
**Pull request by pjenvey**
- move to the latest cloud-storage (w/ a new API)
- upgrade cadence, requiring wrapping of its Client in Arc
- fix the 404 response to use the standard json payload

Closes #386

### CODE REVIEW

File: src/web/handlers.rs [hunk 1]
```diff
@@ -193,7 +193,7 @@ fn content_response(content: &cache::TilesContent) -> HttpResponse {
     match content {
         cache::TilesContent::Json(json) => HttpResponse::Ok()
             .content_type("application/json")
-            .body(json),
+            .body(json.to_owned()),
```
Comment (pjenvey) [inline src/web/handlers.rs:196]: I'll note that actix-web has always internally made a copy of data like `&str` here for response bodies. actix-web 4's API just exposes that to the user, requiring them to make the copy. This makes sense with responses being used in asynchronously, a lifetime would complicate that.

However a `&'static str` can be easily used asynchronously so it supports that and we utilize it w/ `body(EMPTY_TILES.as_str())` (no copying) above.

File: Cargo.toml [hunk 2]
```diff
@@ -15,20 +15,18 @@ edition = "2021"
...
 blake3 = "1.0"
-bytes = "1.0"
-cadence = "0.26"
+bytes = "1"
+cadence = "0.29"
 chrono = "0.4"
 docopt = "1.1"
-cloud-storage = { git = "https://github.com/mozilla-services/cloud-storage-rs", branch = "release/0.6.2-create_with_params" } # 0.7+ includes request 0.11, tokio 1.4
+cloud-storage = { git = "https://github.com/mozilla-services/cloud-storage-rs", branch = "release/0.11.1-client-builder-and-params" }
 config = "0.11"
```
Comment (jrconlin) [inline Cargo.toml:30]: Didn't want to bump this to 0.13?

File: Cargo.toml [hunk 3]
```diff
@@ -15,20 +15,18 @@ edition = "2021"
...
 dashmap = "4.0.2"
```
Comment (jrconlin) [inline Cargo.toml:31]: Looks like bumping this to 5.3 has no code impact

**Review by jrconlin — COMMENTED** [review]
Do we also want to bump sentry up to 0.25?

**Comment by jrconlin** [comment]
Do we also want to bump sentry up to 0.25?

File: Cargo.toml (hunk 2 above)
Comment (pjenvey) [reply Cargo.toml:30]: I was planning to upgrade more things in another PR (like sentry)

**Review by jrconlin — APPROVED** [review]

File: src/adm/settings.rs [hunk 4]
```diff
@@ -320,7 +320,13 @@ impl AdmFilterSettings {
             .timeout(request_timeout)
             .build()
             .map_err(|e| ConfigError::Message(e.to_string()))?;
-        let contents = cloud_storage::Object::download_with(&bucket_name, path, &req)
+        let storage_client = cloud_storage::Client::builder()
```
[the remark this answers is not in the document]
Comment (pjenvey) [reply src/adm/settings.rs]: While collapsing this I ended up with a small refactor which ends up sharing the cloud storage Client a bit more

File: src/server/img_storage.rs [hunk 5]
```diff
@@ -495,14 +507,21 @@ mod tests {
     fn test_store() -> ImageStore {
         let settings = test_storage_settings();
         let timeout = std::time::Duration::from_secs(settings.request_timeout);
+        let req = reqwest::Client::builder()
```
Comment (jrconlin) [inline src/server/img_storage.rs:510]: nit: tempted to rename this `client` instead of `req`.

File: src/server/mod.rs [hunk 6]
```diff
@@ -95,28 +99,31 @@ macro_rules! build_app {
...
     pub async fn with_settings(mut settings: Settings) -> Result<dev::Server, HandlerError> {
-        let metrics = metrics_from_opts(&settings)?;
-        let mut raw_filter = HandlerResult::<AdmFilter>::from(&mut settings)?;
-        // try to update from the bucket if possible.
-        if raw_filter.is_cloud() {
-            raw_filter.update().await?
-        }
-        let filter = Arc::new(RwLock::new(raw_filter));
+        let metrics = Arc::new(metrics_from_opts(&settings)?);
         let req = reqwest::Client::builder()
```
Comment (jrconlin) [inline src/server/mod.rs:110]: nit: tempted to rename this `client`.
We're building a `reqwest::Client`, that is passed to `.client()` and stored as `reqwest_client:`

**Review by jrconlin — APPROVED** [review]

File: src/server/mod.rs (hunk 6 above)
Comment (pjenvey) [reply src/server/mod.rs:110]: Agreed we can clean up this naming, I'll try to tackle it on later PR. I probably wouldn't use `client` now that we also have a `storage_client: cloud_storage::Client`. Maybe `reqwest` and `cloud_storage`?

**Comment by pjenvey** [comment]
I'll wait for #385 before merging this

**Review by jrconlin — APPROVED** [review]

**Pull request merged**